providers/statsig/pkg/provider.go

```
@@ -0,0 +1,328 @@
+	p.status = of.NotReadyState
+}
+
+// provider does not have any hooks, returns empty slice
+func (p *Provider) Hooks() []of.Hook {
+	return []of.Hook{}
+}
+
+// Metadata returns value of Metadata (name of current service, exposed to openfeature sdk)
+func (p *Provider) Metadata() of.Metadata {
+	return of.Metadata{
+		Name: "Statsig",
+	}
+}
+
+func (p *Provider) BooleanEvaluation(ctx context.Context, flag string, defaultValue bool, evalCtx of.FlattenedContext) of.BoolResolutionDetail {
+
+	// TODO to be removed on new SDK version adoption which includes https://github.com/open-feature/spec/issues/238
+	if p.status != of.ReadyState {
+		if p.status == of.NotReadyState {
+			return of.BoolResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewProviderNotReadyResolutionError(providerNotReady),
+					Reason:          of.ErrorReason,
+				},
+			}
+		} else {
+			return of.BoolResolutionDetail{
+				Value: defaultValue,
+				ProviderResolutionDetail: of.ProviderResolutionDetail{
+					ResolutionError: of.NewGeneralResolutionError(generalError),
+					Reason:          of.ErrorReason,
+				},
+			}
+		}
+	}
+
+	statsigUser, err := toStatsigUser(evalCtx)
+	if err != nil {
+		return of.BoolResolutionDetail{
+			Value: defaultValue,
+			ProviderResolutionDetail: of.ProviderResolutionDetail{
+				ResolutionError: of.NewInvalidContextResolutionError(err.Error()),
+				Reason:          of.ErrorReason,
+			},
+		}
+	}
+
+	res := statsig.CheckGate(*statsigUser, flag)
+	return of.BoolResolutionDetail{
+		Value:                    res,
+		ProviderResolutionDetail: of.ProviderResolutionDetail{},
+	}
+}
+
+func (p *Provider) FloatEvaluation(ctx context.Context, flag string, defaultValue float64, evalCtx of.FlattenedContext) of.FloatResolutionDetail {
+	res := p.ObjectEvaluation(ctx, flag, defaultValue, evalCtx)
+	return of.FloatResolutionDetail{
+		Value:                    res.Value.(float64),
```

If it fails type assertion, should it return the default value? Currently, it could silently error

---

updated